x-pack/legacy/plugins/infra/public/containers/logs/log_analysis/log_analysis_status_state.tsx

```diff
@@ -194,6 +269,96 @@ const getJobStatus = (jobId: string) => (jobSummaries: FetchJobStatusResponsePay
       }
     )[0] || 'missing';
 
-export const useStatusState = () => {
-  return useReducer(statusReducer, initialState);
+const getSetupStatus = (
+  spaceId: string,
+  sourceId: string,
+  sourceConfiguration: JobSourceConfiguration,
+  everyJobStatus: Record<JobType, JobStatus>,
+  jobDefinitions: JobDefinition[],
+  jobSummaries: JobSummary[]
+) => (previousSetupStatus: SetupStatus) =>
+  Object.entries(everyJobStatus).reduce<SetupStatus>((setupStatus, [jobType, jobStatus]) => {
+    if (!jobTypeRT.is(jobType)) {
+      return setupStatus;
+    }
+
+    const jobId = getJobId(spaceId, sourceId, jobType);
+    const jobDefinition = jobDefinitions.filter(({ id }) => id === jobType)[0];
+
+    if (jobStatus === 'missing') {
+      return 'required';
+    } else if (
+      setupStatus === 'required' ||
+      setupStatus === 'requiredForUpdate' ||
+      setupStatus === 'requiredForReconfiguration'
+    ) {
+      return setupStatus;
+    } else if (
+      setupStatus === 'skippedButUpdatable' ||
+      (jobDefinition &&
+        !isJobRevisionCurrent(jobId, jobDefinition.config.custom_settings.job_revision || 0)(
+          jobSummaries
+        ))
+    ) {
+      return 'skippedButUpdatable';
+    } else if (
+      setupStatus === 'skippedButReconfigurable' ||
+      !isJobConfigurationConsistent(jobId, sourceConfiguration)(jobSummaries)
+    ) {
+      return 'skippedButReconfigurable';
+    } else if (setupStatus === 'hiddenAfterSuccess') {
+      return setupStatus;
+    } else if (setupStatus === 'skipped' || isJobStatusWithResults(jobStatus)) {
+      return 'skipped';
+    }
+
+    return setupStatus;
+  }, previousSetupStatus);
+
+const isJobRevisionCurrent = (jobId: string, currentRevision: number) => (
+  jobSummaries: FetchJobStatusResponsePayload
+): boolean =>
+  jobSummaries
+    .filter(jobSummary => jobSummary.id === jobId)
+    .every(
+      jobSummary =>
+        jobSummary.fullJob &&
+        jobSummary.fullJob.custom_settings &&
+        jobSummary.fullJob.custom_settings.job_revision &&
+        jobSummary.fullJob.custom_settings.job_revision >= currentRevision
```

Comment:
Is the revision being `>` the current revision something we actually want to mark as being current? I feel like this would suggest something has actually gone wrong, and if they’re not `===` should still acquire a `skippedButUpdatable` status.

Reply:
Good question.

Trying to imagine situations in which this could occur:
* Kibana was downgraded
* The user has an old version of the UI open in the tab.

The downgrade is not supported by Kibana itself, so the UI is probably not shown in that case. The old UI version should also be caught by a Kibana header check, but if that doesn't happen some user might unnecessarily redeploy the same new version.

My original motivation to use `>=` stems from the negation of a `canBeUpgraded` predicate, which would be formulated as `deployed_version < available_version`.

Reply:
Makes sense, thanks for explaining the thought process 👍